### PR TITLE
feat: add new contracts-periphery package

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Introduces a new contracts-periphery package and initializes the package
with the RetroReceiver contract (meant to receive RetroPGF funds on all
available chains). Reason for introducing a new package is that the
original contracts package should be reserved for in-protocol contracts
while contracts-periphery can be used for any contracts that we want to
write and test, regardless of if they're part of the protocol.

### CODE REVIEW

**Comment by tynes** [comment]
Going to need to add this to CI for publishing

**Review by mslipper — APPROVED** [review]

File: packages/contracts-periphery/tasks/deploy-receiver.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      '0x0000000000000000000000000000000000000000000000000000000000000001'
+    const code = hre.ethers.utils.hexConcat([
+      hre.artifacts.readArtifactSync('RetroReceiver').bytecode,
+      hre.ethers.utils.defaultAbiCoder.encode(['address'], [addr]),
+    ])
+
+    // Predict and connect to the contract address
+    const receiver = await hre.ethers.getContractAt(
+      'RetroReceiver',
+      await singleton.callStatic.deploy(code, salt, {
```
Comment (tynes) [inline packages/contracts-periphery/tasks/deploy-receiver.ts:64]: I love this way to get the address to attach to

File: packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+  describe('receive', () => {
+    it('should be able to receive ETH', async () => {
+      await expect(
+        signer1.sendTransaction({
+          to: RetroReceiver.address,
+          value: DEFAULT_AMOUNT,
+        })
+      ).to.not.be.reverted
+    })
```
Comment (tynes) [inline packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts:54]: Would be nice to have an assertion that shows the balance increasing by the expected amount

File: packages/contracts-periphery/contracts/universal/RetroReceiver.sol [hunk 3]
```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
```
Comment (tynes) [inline packages/contracts-periphery/contracts/universal/RetroReceiver.sol:3]: No events?

File: packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+  describe('withdrawETH(address)', () => {
+    describe('when called by the owner', () => {
+      it('should withdraw all ETH in the contract', async () => {
+        await signer1.sendTransaction({
+          to: RetroReceiver.address,
+          value: DEFAULT_AMOUNT,
+        })
+
+        await RetroReceiver['withdrawETH(address)'](DEFAULT_RECIPIENT)
+
```
Comment (tynes) [inline packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts:66]: Could also assert the balance of the receiver goes to 0

File: packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+      await expect(
+        signer1.sendTransaction({
+          to: RetroReceiver.address,
+          value: DEFAULT_AMOUNT,
+        })
+      ).to.not.be.reverted
+    })
+  })
+
+  describe('withdrawETH(address)', () => {
```
Comment (tynes) [inline packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts:57]: This is set up to look like there should be an `onlyOwner` test but I do not see one

File: packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+      })
+    })
+
+    describe('when called by not the owner', () => {
+      it('should revert', async () => {
+        await expect(
+          RetroReceiver.connect(signer2)['withdrawETH(address)'](
+            signer2.address
+          )
+        ).to.be.reverted
```
Comment (tynes) [inline packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts:79]: Is there a way to assert what the revert message is? `Ownable: caller is not the owner`

Comment (tynes) [reply packages/contracts-periphery/test/contracts/universal/RetroReceiver.spec.ts:79]: Unless solmate has a different revert message

**Review by tynes — COMMENTED** [review]
Generally looks good to me, a few nits

File: packages/contracts-periphery/contracts/universal/RetroReceiver.sol (hunk 3 above)
Comment (smartcontracts) [reply packages/contracts-periphery/contracts/universal/RetroReceiver.sol:3]: Added events

**Review by tynes — APPROVED** [review]

**Pull request merged**